hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -1000,13 +1000,22 @@ boolean maybeScaleUp(int dataMembersWithPartitionsCount) {
         return false;
     }
 
-    List<RawJobMetrics> jobMetrics() {
-        return jobMetrics;
+    List<RawJobMetrics> executionMetrics() {
+        return executionMetrics;
     }
 
-    private void setJobMetrics(List<RawJobMetrics> jobMetrics) {
-        assert jobMetrics.stream().allMatch(Objects::nonNull) : "responses=" + jobMetrics;
-        this.jobMetrics = Objects.requireNonNull(jobMetrics);
+    private void setExecutionMetrics(List<RawJobMetrics> executionMetrics) {
+        assert executionMetrics.stream().allMatch(Objects::nonNull) : "responses=" + executionMetrics;
+        this.executionMetrics = Objects.requireNonNull(executionMetrics);
+    }
+
+    private List<RawJobMetrics> withJobMetrics(List<RawJobMetrics> executionMetrics) {
+        List<RawJobMetrics> metrics = new ArrayList<>(executionMetrics.size() + 1);
```

Comment:
this is additional allocation (garbage) and copying. Can we preallocate enough space, so we do not need to resize the array?